### PR TITLE
[now dev] Add `--listen` / `-l` flag, deprecate `--port` / `-p`

### LINKED ISSUE #2704: Prompt to accept incoming connections on Mac OS when running `now dev`
When activating the firewall on MacOS, I start getting these prompts **each time** running `now dev` (sorry for the french):
Capture d’écran 2019-08-05 à 21 24 33

I would expect to accept it only once the first time I run `now dev`, or not at all because I'm not trying to access "internet" or expose a server to the outside world with `now dev`.

One possible root cause for this is that the `now dev` server is listening on `0.0.0.0` instead of `localhost`. Listening on `localhost` wouldn't prompt this message.

Also worth noting that I'm not getting these prompts when running other servers in dev mode (for example, `yarn run next`).

### PR DESCRIPTION
**Pull request by leo**
Deprecates `--port` in favor of a new `--listen` flag, which matches the behavior of `serve`. This allows for binding to a specific network interface, as well as Unix socket files or Windows named pipes.

### CODE REVIEW

**Review by styfle — APPROVED** [review]
I thought @TooTallNate already made this change 🤔

**Comment by TooTallNate** [comment]
@styfle It was reverted in 8638d17d87b425c845c451c03f61dee1db6077a7 because we are going to change it to `--listen` instead.

File: src/util/dev/parse-listen.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    case 'unix:':
+      if (!url.pathname) {
+        throw new Error(`Invalid UNIX domain socket endpoint: ${str}`);
+      }
+
+      return [url.pathname];
+    case 'tcp:':
+      url.port = url.port || String(defaultPort);
+      return [parseInt(url.port, 10), url.hostname];
+    default:
```
Comment (styfle) [inline src/util/dev/parse-listen.ts:38]: ~~Is there a reason why there is no `case 'pipe:'`?~~

Comment (styfle) [reply src/util/dev/parse-listen.ts:38]: Dang I missed it above.

What is this default supposed to be? I don't see it in the unit tests.

Comment (styfle) [reply src/util/dev/parse-listen.ts:38]: Ok I see, its the string with the IP address.

**Review by styfle — APPROVED** [review]

**Pull request merged**